docs/platform-feature-parity.md

```diff
@@ -0,0 +1,159 @@
+# Platform feature-parity
+
+macOS and Windows binaries are new and considered alpha-quality. Some functionality may be missing, some may be broken. (Enhancements and bug-reports welcome, please see [#551: Multi-OS support](https://github.com/aelsabbahy/goss/issues/551))
+
+This matrix attempts to track parity across platforms.
+
+## How to use this doc
+
+### Legend
+
+* `x`    - full supported/tested
+* `w`    - full working/tested, community support
+* `w-nt` - works no automated tests
+* `n/a`  - Not applicable for this platform
+* `#585` - Link to bug(s) for implementation, fixing
+* ` `    - (blank) - not yet tried, no data
+
+### Contributing
+
+The current integration test approach is only appropriate for validating `linux` binaries against `linux` OS/arch combinations.
+
+Validating `macOS` and `Windows` binaries requires adding coverage that runs on those platforms within Travis, but since Travis does not support containerised builds for either platform, assertions are limited to assert against the state of the CI hosts, where we're relying on that to predictable.
+
+## Matrix - tests/assertions
+
+| Test                | `linux` | `macOS` | `Windows` |
+|:--------------------|---------|---------|-----------|
+| **addr**            | x       |         |           |
+| reachable           | x       |         |           |
+| local-address       | x       |         |           |
+| timeout             | x       |         |           |
+|                     | x       |         |           |
+| **command**         | x       |         |           |
+| exit-status         | x       |         |           |
+| stdout              | x       |         |           |
+| stderr              | x       |         |           |
+| timeout             | x       |         |           |
+|                     | x       |         |           |
+| **dns**             | x       |         |           |
+| resolvable          | x       |         |           |
+| addrs               | x       |         |           |
+| server              | x       |         |           |
+| timeout             | x       |         |           |
+| **file**            | x       |         |           |
+| exists              | x       | w-nt    | w         |
+| mode                | x       |         | n/a       |
+| size                | x       |         |           |
+| owner               | x       | #585    | n/a       |
```

Comment:
The #585 were examples.. just points to this PR =). Sorry, I should have clarified the file I created had dummy/example placeholders of how the table values would look. Didn't mean to give the impression that it's actual findings.

Since you've had more experience with using goss multi-os. If you don't mind can you  fill the table to the best of your knowledge?

Reply:
Done inside f7c2d6f.

That was quite a big commit. I:
* added `integration-tests/run-tests-alpha.sh` which
  * finds all the `*.goss.yaml` tests inside `integration-tests`
  * executes `goss validate` against them
* added one file each on Windows per test and some commands
* wired that into `ci/build.sh` so it tries to run in travis
* adjusted `cmd/goss.go` so that the environment var is `GOSS_USE_ALPHA` instead, and wired that into the CLI framework so I could supply a flag within the various self-tests.

That wasn't a small effort, but if you're generally approving I can do the same for macOS when I have a bit more time and some feedback.

I'm also conscious this PR is growing.